django/db/migrations/autodetector.py

```diff
@@ -563,6 +563,19 @@ def generate_created_models(self):
                 if isinstance(base, str) and "." in base:
                     base_app_label, base_name = base.split(".", 1)
                     dependencies.append((base_app_label, base_name, None, True))
+                    # Add dependencies for base model's removed fields that are added into newly
+                    # added sub models so that field removal operations for base models can take
+                    # place before new inherited models are created.
+                    prev_state_base_model = self.from_state.models.get((base_app_label, base_name), 0)
+                    curr_state_base_model = self.to_state.models.get((base_app_label, base_name), 0)
+                    if prev_state_base_model and curr_state_base_model:
+                        prev_base_fields = set(field[0] for field in prev_state_base_model.fields)
+                        curr_base_fields = set(field[0] for field in curr_state_base_model.fields)
+                        removed_base_fields = prev_base_fields.difference(curr_base_fields)
+                        curr_model_fields = set(field[0] for field in model_state.fields)
```

Comment:
It's clearer to unpack the tuples rather than index with 0

```suggestion
                        prev_base_fields = set(name for name, field in prev_state_base_model.fields)
                        curr_base_fields = set(name for name, field in curr_state_base_model.fields)
                        removed_base_fields = prev_base_fields.difference(curr_base_fields)
                        curr_model_fields = set(name for name, field in model_state.fields)
```

Reply:
that makes a lot more sense!